### ISSUE TITLE
[Data catalog] Move `proximity.dist_pin_to_pin` CTAS query into the dbt DAG

### ISSUE THREAD

**Issue by jeancochrane** [ccao-data/data-architecture#122]
While we moved just about all of the CTAS queries into the DAG in https://github.com/ccao-data/data-architecture/pull/111, we left out `proximity.dist_pin_to_pin` because it requires a more complex solution. Create a separate PR to move `proximity.dist_pin_to_pin` into the DAG by defining a series of intermediate transformations that gradually expand the radius of the buffer.

See this comment thread for more context: https://github.com/ccao-data/data-architecture/pull/111#discussion_r1321844067

Blocked by #111.
